### PR TITLE
Implements #557 - Add IPv4 /32 and IPv6 /128 prefix support

### PR DESCRIPTION
**Pull request by evdzande**
### Fixes: #557 - Add IPv4 /32 and IPv6 /128 prefix support

Allow /32 and /128 prefixes

### CODE REVIEW

**Comment by evdzande** [comment]
this is the first time I created a PR so hopefully done it the right way ;-)

**Comment by glennmatthews** [comment]
Thank you for the pull request! Just to clarify, is this a port of https://github.com/netbox-community/netbox/pull/6545?

**Comment by evdzande** [comment]
@glennmatthews that is correct

**Comment by glennmatthews** [comment]
Thank you again! It looks like the fix for https://github.com/netbox-community/netbox/issues/6350 (https://github.com/netbox-community/netbox/pull/6371/commits/7f2f98885b68b9b9d7208587df3f277c22806b2b) also touches this same code - would you be willing to include that fix in this PR as well for completeness' sake?

**Comment by evdzande** [comment]
Added and tested the fix from  netbox-community/netbox#6350

File: nautobot/ipam/models.py [hunk 1]
```diff
@@ -661,17 +651,18 @@ def get_available_ips(self):
         child_ips = netaddr.IPSet([ip.address.ip for ip in self.get_child_ips()])
         available_ips = prefix - child_ips
 
-        # All IP addresses within a pool are considered usable
-        if self.is_pool:
+        # IPv6, pool, or IPv4 /31 sets are fully usable
+        if self.family == 6 or self.is_pool or self.prefix.prefixlen == 31:
             return available_ips
```
Comment (glennmatthews) [inline nautobot/ipam/models.py:656]: Looking at https://github.com/netbox-community/netbox/pull/6545/files and https://github.com/netbox-community/netbox/commit/7f2f98885b68b9b9d7208587df3f277c22806b2b in context, I think the below might be preferable:

```suggestion
        # IPv6, pool, or IPv4 /31-32 sets are fully usable
        if self.family == 6 or self.is_pool or (self.family == 4 and self.prefix.prefixlen >= 31):
            return available_ips
```

File: nautobot/ipam/models.py [hunk 2]
```diff
@@ -661,17 +651,18 @@ def get_available_ips(self):
...
 
-        # All IP addresses within a point-to-point prefix (IPv4 /31 or IPv6 /127) are considered usable
-        if (self.prefix.version == 4 and self.prefix.prefixlen == 31) or (  # RFC 3021
-            self.prefix.version == 6 and self.prefix.prefixlen == 127  # RFC 6164
+        # All IP addresses within a prefix are considered usable
+        if (self.prefix.version == 4 and self.prefix.prefixlen >= 31) or (  # RFC 3021
+            self.prefix.version == 6 and self.prefix.prefixlen >= 127  # RFC 6164
         ):
             return available_ips
```
Comment (glennmatthews) [inline nautobot/ipam/models.py:662]: If the above change is made, then I think this block can be removed entirely.

```suggestion
```

**Comment by evdzande** [comment]
checked and tested your suggestions. They are working as expected. Thanks!

**Review by glennmatthews — APPROVED** [review]
Thank you for this contribution!  We recognize the fact this code was initially contributed to NetBox by @crafty-ua in PR https://github.com/netbox-community/netbox/pull/6545 and by @jeremystretch in https://github.com/netbox-community/netbox/commit/7f2f98885b68b9b9d7208587df3f277c22806b2b.  Thank you for the contribution to Nautobot and thank you @crafty-ua and @jeremystretch again for the initial development of the code.

**Pull request merged**